js/web/lib/wasm/jsep/webgpu/ops/pool.ts

```diff
@@ -312,16 +368,24 @@ const createMaxPoolProgramInfo =
       value = max(x_val, value);
     `;
       const op2 = '';
-      const x = inputVariable('x', input.dataType, input.dims);
+      const x = inputVariable('x', input.dataType, input.dims.length);
+      const inputDependencies: ProgramInputTensorInfoDependency[] = ['rank'];
+      const start = -1e5;
+      const [programUniforms, uniforms, hasPads, pwStartEnd, phStartEnd] =
+          getUniformAndPadInfo(input.dims, outputShape, x.type.value, start, 'int32', adjustedAttributes);
+      programUniforms.push(...createTensorShapeVariables(input.dims));
+      programUniforms.push(...createTensorShapeVariables(outputShape));
       return {
         name,
-        shaderCache: {hint: attributes.cacheKey},
+        shaderCache: {hint: attributes.cacheKey + op1 + op2 + hasPads, inputDependencies},
```

Comment:
`attributes.cacheKey` will include all the constant attributes information, such as `"NOTSET;0;false;NHWC;3,3;2,2,2,2;1,1"`. Since you have used uniform for most of them. Please only include the necessary key value. Otherwise, it still generates many shader variants.

Reply:
remove + op1 + op2